packages/slo-stat-bot/src/slo-appliesTo.ts

```diff
@@ -0,0 +1,210 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// eslint-disable-next-line node/no-extraneous-import
+import {SLORules} from './types';
+
+/**
+ * Function determines if the type of issue applies to slo
+ * @param issues slo rule if it applies to issues
+ * @param prs slo rule if it applies to prs
+ * @param type specifies if event is issue or pr
+ * @returns true if type applies to issues else false
+ */
+export const isValidType = async function isValidType(
+  issues: boolean | undefined,
+  prs: boolean | undefined,
+  type: string
+): Promise<boolean> {
+  issues = issues === undefined ? true : issues;
+  prs = prs === undefined ? false : prs;
+
+  if (type === 'pull_request' && prs) {
+    return true;
+  }
+  if (type === 'issue' && issues) {
+    return true;
+  }
+  return false;
+};
+
+/**
+ * Function checks if all the githublabels are subset of issue labels
+ * @param issueLabels of the issue
+ * @param githubLabels is slo rule for github labels that must exist in issue
+ * @returns true if githubLabels applies to issues else false
+ */
+export const isValidGithubLabels = async function isValidGithubLabels(
```

Comment:
these seem like very small functions. Is it possible to make them inline in your code? Might make it easier to read! It sounds like you could just do something like 
```if(!githubLabels) {
//code I want to execute
}
```
instead of returning true to then cause something else to happen. Just a thought!

Reply:
I think the same goes for ```isValidExcludedLabels```, ```isElementExist```, etc.

Reply:
That is a fair point. Initially I wanted to make every function isolated so it would be easier to understand the decision tree and the reasoning for a specific outcome. But it does result in extra code and space. So I can try condensing it down.

Reply:
I also wanted to make determining if it is `isValidGithubLabels`, `isValidExcludedLabels`, `isElementExist` more faster by making those functions asynchronous. If the slo does not apply to any one of the given functions it would immediately return false instead of waiting for the other lines to execute. But considering these are much simpler functions I don't think that should be a problem since it should compute fast.